python/lsst/ip/isr/photodiode.py

```diff
@@ -0,0 +1,260 @@
+# This file is part of ip_isr.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+Photodiode storage class.
+"""
+import numpy as np
+from astropy.table import Table
+
+from lsst.ip.isr import IsrCalib
+
+
+__all__ = ["PhotodiodeCalib"]
+
+
+class PhotodiodeCalib(IsrCalib):
+    """Independent current measurements from photodiode for linearity
+    calculations.
+
+    Parameters
+    ----------
+    timeSamples : `list` or `numpy.ndarray`
+        List of samples the photodiode was measured at.
+    currentSamples : `list` or `numpy.ndarray`
+        List of current measurements at each time sample.
+    log : `lsst.log.Log`, optional
+        Log to write messages to.
+    **kwargs :
+        Paramters to pass to parent constructor.
+    """
+
+    _OBSTYPE = 'PHOTODIODE'
+    _SCHEMA = 'Gen3 Photodiode'
```

Comment:
I thought we were trying not to say Gen3, but you know a lot more about this than I do, so ignore as needed.